src/generator/AutoRest.Java.Azure.Fluent/Model/MethodJvaf.cs

```diff
@@ -27,34 +28,49 @@ public override void Disambiguate()
 
             var methodType = GetMethodType(this);
             var originalName = Name;
-            string newName = null;
 
-            if (methodType == MethodType.ListBySubscription)
+            if (methodType != MethodType.Other)
             {
-                var otherListBySubMethods = this.MethodGroup.Methods.Where(x => GetMethodType(x as MethodJvaf) == MethodType.ListBySubscription);
-                if (otherListBySubMethods.Count() == 1)
+                string newName = null;
+
+                if (this.MethodGroup.Methods.Count(x => GetMethodType(x as MethodJvaf) == methodType) == 1)
                 {
-                    newName = "List";
+                    switch (methodType)
+                    {
+                        case MethodType.ListBySubscription:
+                            newName = WellKnowMethodNames.List;
+                            break;
+
+                        case MethodType.ListByResourceGroup:
+                            newName = WellKnowMethodNames.ListByResourceGroup;
+                            break;
+
+                        case MethodType.Delete:
+                            newName = WellKnowMethodNames.Delete;
+                            break;
+
+                        case MethodType.Get:
+                            newName = WellKnowMethodNames.GetByResourceGroup;
```

Comment:
Results into
https://github.com/Azure/azure-sdk-for-java/pull/1507/files#diff-58b8ce8b505f5a743c899ad5501964b8R71